mox/impl/src/lib.rs

```diff
@@ -1,245 +1,239 @@
 extern crate proc_macro;
 
-use proc_macro2::{Ident, Span, TokenStream, TokenTree};
-use proc_macro_error::{abort, emit_error, proc_macro_error, Diagnostic, Level, ResultExt};
-use proc_macro_hack::proc_macro_hack;
+use proc_macro2::{Span, TokenStream};
 use quote::{quote, ToTokens};
-use snax::{ParseError, SnaxAttribute, SnaxFragment, SnaxItem, SnaxSelfClosingTag, SnaxTag};
-use std::borrow::Cow;
+use std::convert::TryFrom;
+use syn::{parse::Parse, parse_macro_input, spanned::Spanned};
 
-#[proc_macro_error(allow_not_macro)]
-#[proc_macro_hack]
+#[proc_macro]
 pub fn mox(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
-    let item = snax::parse(input.into()).map_err(Error::SnaxError).unwrap_or_abort();
-    let item = MoxItem::from(item);
+    let item = parse_macro_input!(input as MoxItem);
     quote!(#item).into()
 }
 
 enum MoxItem {
```

Comment:
Can you organize this file so that types' definitions are with their impls? Something like

```
enum MoxItem { ... }
impl Parse for MoxItem { ... }
impl TryFrom<RsxNode> for MoxItem { ... }
impl ToTokens for MoxItem { ... }

struct MoxTag { ... }
impl TryFrom<RsxNode> for MoxTag { ... }
impl ToTokens for MoxTag { ... }

struct MoxAttr { ... }
impl TryFrom<RsxNode> for MoxAttr { ... }
impl ToTokens for MoxAttr { ... }

struct MoxExpr { ... }
impl TryFrom<RsxNode> for MoxExpr { ... }
impl ToTokens for MoxExpr { ... }
```

Reply:
Without line breaks between? Well ok

Reply:
I will hold it for now to not outdate other questions.

Reply:
> Without line breaks between? Well ok

Sorry for ambiguity, not without linebreaks in between. I left linebreaks out to clarify the grouping I meant, sorry.

Good call on holding off for other questions!